guides/source/caching_with_rails.md

```diff
@@ -286,90 +349,196 @@ simply be explicit in a comment, like:
-    @products = Product.all
-  end
-end
+### Handling Cache Expiration
+
+Solid Cache tracks cache writes by incrementing a counter with each write. When the counter reaches 50% of the `expiry_batch_size` from the [Cache configuration](https://github.com/rails/solid_cache#cache-configuration), a background task is triggered to handle cache expiry. This approach ensures cache records expire faster than they are written when the cache needs to shrink.
+
+The background task only runs when there are writes, so the process stays idle when the cache is not being updated. If you prefer to run the expiry process in a background job instead of a thread, set `expiry_method` from the[Cache configuration](https://github.com/rails/solid_cache#cache-configuration) to `:job`, which will enqueue a `SolidCache::ExpiryJob`.
+
+### Sharding the Cache
+
+If you need more scalability, Solid Cache supports sharding — splitting the cache across multiple databases. This spreads the load, making your cache even more powerful. To enable sharding, add multiple cache databases to your database.yml:
+
+```yaml
+# config/database.yml
+production:
+  cache_shard1:
+    database: cache1_production
+    host: cache1-db
+  cache_shard2:
+    database: cache2_production
+    host: cache2-db
+  cache_shard3:
+    database: cache3_production
+    host: cache3-db
 ```
 
-The second time the same query is run against the database, it's not actually going to hit the database. The first time the result is returned from the query it is stored in the query cache (in memory) and the second time it's pulled from memory.
+Additionally, you must specify the shards in the cache configuration:
 
-However, it's important to note that query caches are created at the start of
-an action and destroyed at the end of that action and thus persist only for the
-duration of the action. If you'd like to store query results in a more
-persistent fashion, you can with low-level caching.
+```yaml
+# config/cache.yml
+production:
+  databases: [cache_shard1, cache_shard2, cache_shard3]
+```
+
+### Encryption
 
-Cache Stores
-------------
+Solid Cache supports encryption to protect sensitive data. To enable encryption, set the `encrypt` value in your cache configuration:
 
-Rails provides different stores for the cached data (apart from SQL and page
-caching).
+```yaml
+# config/cache.yml
+production:
+  encrypt: true
+```
+
+You will need to set up your application to use[Active Record Encryption](active_record_encryption.html).
+
+### Caching in Development
+
+By default, caching is *enabled* in development mode with
+[`:memory_store`](#activesupport-cache-memorystore).
+This doesn't apply to Action Controller caching, which is disabled
+by default.
+
+To enable Action Controller caching Rails provides the `bin/rails dev:cache` command.
+
+```bash
+$ bin/rails dev:cache
+Development mode is now being cached.
+$ bin/rails dev:cache
+Development mode is no longer being cached.
+```
```

Comment:
This isn't specific to Solid Cache, maybe move it under "Types of Caching" where we also mention `bin/rails dev:cache`?